test/fluree/db/query/misc_queries_test.clj

```diff
@@ -390,3 +391,174 @@
+                                               "ex:note" "Another test"}
+                                              {"@id" "ex:newData3"
+                                               "ex:name" "Third Item"}
+                                              {"@id" "ex:newData5"
+                                               "ex:description" "Fifth Item"}
+                                              {"@id" "ex:newData6"
+                                               "ex:label" "Sixth Item"
+                                               "ex:comment" "Last item"}]})]
+
+      ;; Test the complex query with values, datatype bind, and filter
+      (let [query {"values" ["?s"
+                             [{"@type" "@id", "@value" "http://example.org/existingOrNewData"}
+                              {"@type" "@id", "@value" "http://example.org/newData"}
+                              {"@type" "@id", "@value" "http://example.org/newData2"}
+                              {"@type" "@id", "@value" "http://example.org/newData3"}
+                              {"@type" "@id", "@value" "http://example.org/newData5"}
+                              {"@type" "@id", "@value" "http://example.org/newData6"}]]
+                   "select" ["?s" "?property" "?value"]
+                   "groupBy" ["?s" "?property"]
+                   "where" [{"@id" "?s"
+                             "?property" "?value"}
+                            ["bind" "?dt" "(datatype ?value)"]
+                            ["filter" "(or (= ?dt \"http://www.w3.org/2001/XMLSchema#string\") (= ?property (iri \"@type\")))"]]
+                   "@context" {"ex" "http://example.org/"}}
+            result @(fluree/query db query)]
+
+        ;; Verify we got results
+        (is (pos? (count result))
+            "Should return results for string properties")
+
+        ;; All results should have string values or be @type predicates
+        ;; Note: groupBy wraps values in arrays
+        (is (every? #(let [value (nth % 2)]
+                       (or (and (vector? value)
+                                (= 1 (count value))
+                                (string? (first value)))
+                           (= "http://www.w3.org/1999/02/22-rdf-syntax-ns#type" (nth % 1))))
+                    result)
+            "All values should be strings (wrapped in arrays due to groupBy) or the property should be rdf:type")
+
+        ;; Check that we're getting results for our test subjects
+        (let [subjects (set (map first result))]
+          (is (some #(str/ends-with? % "existingOrNewData") subjects)
+              "Should include existingOrNewData in results")
+          (is (some #(str/ends-with? % "newData") subjects)
+              "Should include newData in results"))
+
+        ;; Verify the datatype function and filter worked by checking we only get string values
+        (let [property-values (filter #(not= "http://www.w3.org/1999/02/22-rdf-syntax-ns#type" (nth % 1)) result)]
```

Comment:
This can be consolidated with the two tests right above this and made clearer with a named let binding.

Reply:
Also I think we should be returning `@type` instead of `http://www.w3.org/1999/02/22-rdf-syntax-ns#type`.